File: HermesSenaProyect-frontend/hermes-frontend/src/app/model/delivery-response.ts
import { DeliveryProductResponse } from "./delivery-product-response";

export class DeliveryResponse {
    id: number;
    nameCustomer: string;
    addressCustomer: string;
    phoneCustomer: string;
    deliveryProduct: DeliveryProductResponse[];
    dateOfDelivery: string;
    totalOfAll: number;
    status: boolean;

    constructor(id: number, nameCustomer: string, addressCustomer: string, phoneCustomer: string, deliveryProduct: DeliveryProductResponse[], dateOfDelivery: string, totalOfAll: number, status: boolean) {
        this.id = id;
        this.nameCustomer = nameCustomer;
        this.addressCustomer = addressCustomer;
        this.phoneCustomer = phoneCustomer;
        this.deliveryProduct = deliveryProduct;
        this.dateOfDelivery = dateOfDelivery;
        this.totalOfAll = totalOfAll;
        this.status = status;
    }
}

File: hermes.of.cheese.backend/src/main/java/com/sena/proyect/hermes/of/cheese/presentation/controller/response/DeliveryResponse.java
package com.sena.proyect.hermes.of.cheese.presentation.controller.response;

import com.sena.proyect.hermes.of.cheese.persistence.entity.DeliveryProduct;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class DeliveryResponse {
    private Long id;
    private String nameCustomer;
    private String addressCustomer;
    private String phoneCustomer;
    private List<DeliveryProductResponse> deliveryProduct = new ArrayList<>();
    private LocalDateTime dateDelivery;

    private Double totalOfAll = 0.0;
    private boolean status;

}
